From c71796661b00505f42170d267b196ab0fcbcef1e Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Wed, 13 Jul 2016 16:54:56 -0400
Subject: [PATCH 0644/1722] drm/amd/dal: bump displayclock up by 15 percent

This works around a vbios bug where they round down when calculating
dividers, setting a displayclock to a lower value than required.
1080p video multiplane shows jagged edges because of this.

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index 62a2cf2..bdee9f1 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1028,9 +1028,18 @@ static void set_display_clock(struct validate_context *context)
 				path_set,
 				context->res_ctx.pool.display_clock,
 				context->res_ctx.min_clocks.min_dclk_khz);*/
-	} else
-		dal_display_clock_set_clock(context->res_ctx.pool.display_clock,
-				context->bw_results.dispclk_khz);
+        } else {
+                /*
+                 * TODO: need to either port work around from DAL2 function
+                 * getActualRequiredDisplayClock or program displayclock without
+                 * calling vbios. Currently temporily work
+                 * around by increasing the displclk by 15 percent
+                 */
+                dal_display_clock_set_clock(
+                                context->res_ctx.pool->display_clock,
+                                context->bw_results.dispclk_khz * 115 / 100);
+        }
+
 
 	/* TODO: When changing display engine clock, DMCU WaitLoop must be
 	 * reconfigured in order to maintain the same delays within DMCU
-- 
2.7.4

